src/utils/fetchUtils.ts

```diff
@@ -224,3 +231,62 @@ export async function fetchLanguages(): Promise<ILanguage[]> {
   });
   return languages;
 }
+
+function sortByHourAndMin(a: IEvent, b: IEvent) {
+  const aStart = a.dateStart;
+  const bStart = b.dateStart;
+  if (aStart.getHours() - bStart.getHours() === 0) {
+    return aStart.getMinutes() - bStart.getMinutes();
+  }
+  return aStart.getHours() - bStart.getHours();
+}
+
+// Since some dates in API have actual hours and others are always at 00:00
+// (meaning entire days presumably), we compare by start and end of day
+function isIncludedInDateRange(start: string, end: string) {
+  const rangeStartDate = startOfDay(new Date(start));
+  const rangeEndDate = endOfDay(new Date(end));
+  return (item: IEvent) => {
+    const { dateStart, dateEnd } = item;
+    const zonedStart = startOfDay(zonedTimeToUtc(dateStart, TZ));
+    const zonedEnd = endOfDay(zonedTimeToUtc(dateEnd, TZ));
+    return zonedStart <= rangeStartDate && zonedEnd >= rangeEndDate;
+  };
+}
+
+export async function fetchEvents(
+  userGroupId: number,
+  lang?: string,
+  dateStart?: string,
+  dateEnd?: string,
+): Promise<IEvent[]> {
+  const url = buildEventsUrl(userGroupId, lang, dateStart, dateEnd);
+  logApp(`sending fetch request to: ${url}`);
+
+  // TODO: refactor this?
+  const response = await fetch(url);
+  logApp(`received fetching response from: ${url}`);
+  if (!response.ok) {
+    throw new Error("Malformed request");
+  }
+
+  let events: IEvent[] = [];
+  const eventsJson = await response.json();
+  eventsJson.forEach((data: any) => {
```

Comment:
Maybe we should try and type `data` instead of providing `any` here?

Reply:
All the functions in `parseUtils` have `data: any` so I'm thinking there's a reason for that 😄